hack/generator/pkg/astmodel/property_definition.go

```diff
@@ -55,11 +56,26 @@ func (property *PropertyDefinition) PropertyType() Type {
 
 // SetRequired sets if the property is required or not
 func (property *PropertyDefinition) SetRequired(required bool) *PropertyDefinition {
+	if required == property.isRequired {
+		return property
+	}
+
 	result := *property
```

Comment:
This should use `.copy()` to safely copy the `tags` map as well.

Reply:
do we still need to copy it even though we aren't modifying the tags?